src/plans/stopping_criterion.jl

```diff
@@ -564,7 +659,8 @@ function (sc::StopWhenGradientNormLess)(
         sc.at_iteration = -1
     end
     if (k > 0)
-        sc.last_change = sc.norm(M, get_iterate(s), get_gradient(s))
+        r = (has_components(M) && !ismissing(sc.outer_norm)) ? [sc.outer_norm] : []
```

Comment:
```suggestion
        r = (has_components(M) && !ismissing(sc.outer_norm)) ? (sc.outer_norm,) : ()
```
This should be faster.